src/Updater.php

```diff
@@ -177,20 +172,72 @@ public function executeUpdate($module, $update_definition_name) {
         $new_config = NestedArray::mergeDeep($new_config, $update_actions['add']);
       }
 
-      if ($this->updateConfig($configName, $new_config, $expected_config, $delete_keys)) {
+      if ($this->updateConfig($configName, $new_config, $configChange['expected_config'], $delete_keys)) {
         $this->logger->info($this->t('Configuration @configName has been successfully updated.', ['@configName' => $configName]));
       }
       else {
-        $successful = FALSE;
-        $this->logger->warning($this->t('Unable to update configuration for @configName.', ['@configName' => $configName]));
+        $this->registerWarning($this->t('Unable to update configuration for @configName.', ['@configName' => $configName]));
       }
     }
+  }
 
-    // Dispatch event after update has finished.
-    $event = new ConfigurationUpdateEvent($module, $update_definition_name, $successful);
-    $this->eventDispatcher->dispatch(UpdateHelperEvents::CONFIGURATION_UPDATE, $event);
+  /**
+   * Register warning with logging.
+   *
+   * @param string $message
+   *   The message used for logging of warning.
+   */
+  protected function registerWarning($message) {
+    $this->warningCount++;
 
-    return $successful;
+    $this->logger->warning($message);
+  }
+
+  /**
+   * Installs modules.
+   *
+   * @param array $modules
+   *   List of module names.
+   */
+  protected function installModules(array $modules) {
+    foreach ($modules as $module) {
+      try {
+        if ($this->moduleInstaller->install([$module])) {
```

Comment:
One thing that is always interesting about installing modules is that that can rebuild the container. So at this point odd things can happen if we don't re-inject services. See \Drupal\Core\Config\ConfigImporter::reInjectMe() for something we should consider copying for now.

Reply:
Another thing that is always problematic is that you can end up installing more than one module because of dependencies. That's not always desirable - sometimes in updates you are trying to fix things so want to ignore the dependency tree. Not sure what this means for us but it is worth thinking about.

Reply:
I'm still not sure about introducing `reInjectMe` here. We can talk during daily about that.

Reply:
@alexpott for now I would keep module installation functionality as it is. I would not add `reInjectMe`, for that I would create follow-up ticket and address it there.

In this issue we are trying just to simplify API. So what we have before in `installModules`, is now executed by processing information provided in update definition file and that's it.